internal/provider/schema_gitlab_project_milestone.go

```diff
@@ -0,0 +1,127 @@
+package provider
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/xanzy/go-gitlab"
+)
+
+func gitlabProjectMilestoneGetSchema() map[string]*schema.Schema {
+	validMilestoneStates := []string{"active", "closed"}
+
+	return map[string]*schema.Schema{
+		"project": {
+			Description: "The ID or URL-encoded path of the project owned by the authenticated user.",
+			Type:        schema.TypeString,
+			ForceNew:    true,
+			Required:    true,
+		},
+		"title": {
+			Description: "The title of a milestone.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"description": {
+			Description: "The description of the milestone.",
+			Type:        schema.TypeString,
+			Optional:    true,
+		},
+		"due_date": {
+			Description:      "The due date of the milestone. Date time string in the format YYYY-MM-DD, for example 2016-03-11.",
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: isISO6801Date,
+		},
+		"start_date": {
+			Description:      "The start date of the milestone. Date time string in the format YYYY-MM-DD, for example 2016-03-11.",
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: isISO6801Date,
+		},
+		// NOTE: not part of `CREATE`, but part of `UPDATE` with the `state_event` field.
+		"state": {
+			Description:      fmt.Sprintf("The state of the milestone. Valid values are: %s.", renderValueListForDocs(validMilestoneStates)),
+			Type:             schema.TypeString,
+			Optional:         true,
+			Default:          "active",
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validMilestoneStates, false)),
+		},
+		"created_at": {
+			Description: "The time of creation of the milestone. Date time string, ISO 8601 formatted, for example 2016-03-11T03:45:40Z.",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			// NOTE: since RFC3339 is pretty much a subset of ISO8601 and actually expected by GitLab,
+			//       we use it here to avoid having to parse the string ourselves.
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IsRFC3339Time),
```

Comment:
According to the API [docs](https://docs.gitlab.com/ee/api/milestones.html#edit-milestone), `created_at` cannot be set by the user.

```suggestion
			Computed:    true,
```